src/diff/base.js

```diff
@@ -144,8 +144,15 @@ Diff.prototype = {
   },
 
   equals(left, right) {
-    return left === right
-      || (this.options.ignoreCase && left.toLowerCase() === right.toLowerCase());
+    if (this.options.comparator) {
+      if (Array.isArray(right)) {
+        right = right[0];
+      }
```

Comment:
What is this array lookup doing? Is there a case where arrays are passed in here?

Reply:
Yes there is some edge case - maybe because of how this lib uses `tokenize()` and `join()`

    arrayDiff.tokenize = arrayDiff.join = function(value) {
      return value.slice();
    };